src/desktopfilereader.cpp

```diff
@@ -137,11 +137,12 @@ namespace linuxdeploy {
                                     }
 
                                     // locale may only contain A-Za-z- characters according to specification
-                                    for (const char c : entryName) {
+                                    for (const char c : entryLocale) {
                                         if (!(
                                             (c >= 'A' && c <= 'Z') ||
                                             (c >= 'a' && c <= 'z') ||
-                                            (c == '_') || (c == '@')
+                                            (c == '_') || (c == '@') ||
+                                            (c == '[') || (c == ']')
```

Comment:
This permits the use of `[` characters inside the locale part. I think you should remove this line, and iterate over the inner part only.

Reply:
Actually, no, this is handled elsewhere already. However, I find it confusing to permit these here. I am going to update your branch to iterate only over the inner part of the string.